src/main/java/com/microsoft/cse/helium/app/dao/ActorsDao.java

```diff
@@ -11,25 +12,78 @@
 import com.microsoft.azure.spring.data.cosmosdb.core.convert.ObjectMapperFactory;
 import com.microsoft.cse.helium.app.models.Actor;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
 import reactor.core.publisher.Flux;
-// import reactor.core.publisher.Mono;
+import reactor.core.publisher.Mono;
+
+import javax.xml.ws.soap.AddressingFeature;
+import java.util.regex.Pattern;
+
 
 @Service
 public class ActorsDao extends BaseCosmosDbDao {
 
+    private static final Logger _logger = LoggerFactory.getLogger(ActorsDao.class);
+
     final String _actorSelect = "select m.id, m.partitionKey, m.actorId, m.type, m.name, m.birthYear, m.deathYear, m.profession, m.textSearch, m.movies from m where m.type = 'Actor' ";
     final String _actorContains = " and contains(m.textSearch, \"%s\") ";
     final String _actorOrderBy = " order by m.name ";
     final String _actorOffset = " offset %d limit %d ";
 
+    final String _actorSelectById = _actorSelect + " and m.actorId = '%s'";
+
+
     @Autowired
     public ActorsDao(IConfigurationService configService){
         super(configService);
     }
 
+    private Boolean validActorId(String _actorId){
+        /* Valid input: starts with 'nm' (case sensitive)
+        followed by 5-9 digits
+        parses to int > 0
+        */
+        System.out.println(_actorId);
```

Comment:
the underscore in the actorId is not necessary and is not per the Java coding recommendations on Wikipedia.

Reply:
Yes good point, we have another task for following nomenclature , will take it in that PR